src/measure_theory/constructions/polish.lean

```diff
@@ -0,0 +1,689 @@
+  {f : α → β} (f_cont : continuous f) (f_inj : injective f) :
+  measurable_embedding f :=
+{ injective := f_inj,
+  measurable := f_cont.measurable,
+  measurable_set_image' := λ u hu,
+    hu.image_of_continuous_on_inj_on f_cont.continuous_on (f_inj.inj_on _) }
+
+/-- If `s` is Borel-measurable in a Polish space and `f` is continuous injective on `s`, then
+the restriction of `f` to `s` is a measurable embedding. -/
+theorem _root_.continuous_on.measurable_embedding {α : Type*} [topological_space α]
+  [polish_space α] [measurable_space α] [borel_space α]
+  {β : Type*} [topological_space β] [t2_space β] [measurable_space β] [borel_space β] {s : set α}
+  (hs : measurable_set s) {f : α → β} (f_cont : continuous_on f s) (f_inj : inj_on f s) :
+  measurable_embedding (s.restrict f) :=
+{ injective := inj_on_iff_injective.1 f_inj,
+  measurable := (continuous_on_iff_continuous_restrict.1 f_cont).measurable,
+  measurable_set_image' :=
+  begin
+    assume u hu,
+    have A : measurable_set ((coe : s → α) '' u) :=
+      (measurable_embedding.subtype_coe hs).measurable_set_image.2 hu,
+    have B : measurable_set (f '' ((coe : s → α) '' u)) :=
+      A.image_of_continuous_on_inj_on (f_cont.mono (subtype.coe_image_subset s u))
+        (f_inj.mono ((subtype.coe_image_subset s u))),
+    rwa ← image_comp at B,
+  end }
+
+/-- An injective measurable function from a Polish space to a second-countable topological space
+is a measurable embedding. -/
+theorem _root_.measurable.measurable_embedding {α : Type*} [topological_space α]
+  [polish_space α] [measurable_space α] [borel_space α]
+  {β : Type*} [topological_space β] [t2_space β] [second_countable_topology β]
+  [measurable_space β] [borel_space β]
+  {f : α → β} (f_meas : measurable f) (f_inj : injective f) :
+  measurable_embedding f :=
+{ injective := f_inj,
+  measurable := f_meas,
+  measurable_set_image' := λ u hu, hu.image_of_measurable_inj_on f_meas (f_inj.inj_on _) }
+
+/-- In a Polish space, a set is clopenable if and only if it is Borel-measurable. -/
+lemma is_clopenable_iff_measurable_set {α : Type*} [t : topological_space α]
+  [polish_space α] [measurable_space α] [borel_space α] {s : set α} :
+  is_clopenable s ↔ measurable_set s :=
+begin
+  -- we already know that a measurable set is clopenable. Conversely, assume that `s` is clopenable.
+  refine ⟨λ hs, _, λ hs, hs.is_clopenable⟩,
+  -- consider a finer topology `t'` in which `s` is open and closed.
+  obtain ⟨t', t't, t'_polish, s_closed, s_open⟩ :
+    ∃ (t' : topological_space α), t' ≤ t ∧ @polish_space α t' ∧ @is_closed α t' s ∧
+      @is_open α t' s := hs,
+  -- the identity is continuous from `t'` to `t`.
+  have C : @continuous α α t' t id := continuous_id_of_le t't,
+  -- therefore, it is also a measurable embedding, by the Lusin-Souslin theorem
+  have E := @continuous.measurable_embedding α t' t'_polish (@borel α t') (by { constructor, refl })
+    α t (polish_space.t2_space α) _inst_3 _inst_4 id C injective_id,
```

Comment:
I think that you should give names to the instances instead of using `_inst_3`. This breaks too easily when changing unrelated things elsewhere in the file.